mxcubeweb/core/models/configmodels.py

```diff
@@ -19,7 +20,9 @@ class FlaskConfigModel(BaseModel):
     ALLOWED_CORS_ORIGINS: list[str] = Field([], description="")
     SECURITY_PASSWORD_SALT: str = Field("ASALT", description="")
     SECURITY_TRACKABLE: bool = Field(True, description="")
-    USER_DB_PATH: str = Field("/tmp/mxcube-user.db", description="")
+    USER_DB_PATH: str = Field(
+        str(Path.home() / ".mxcube" / "tmp" / "mxcube-user.db"), description=""
```

Comment:
Alright... okay... now we are shifting gears! :D

Then at this point I would suggest using conventional locations like the ones specified in "[XDG Base Directory](https://wiki.archlinux.org/title/XDG_Base_Directory)". So I guess we want `XDG_DATA_HOME`. Concretely, I would recommend using the [`platformdirs`](https://pypi.org/project/platformdirs/) library and in particular the output of `platformdirs.user_data_dir("mxcube")` as base directory. I guess that would be a good default.

Reply:
Hm, I think I would prefer `~./mxcube/tmp` or `~./mxcube/` over the ie. `~./config/`. `USER_DB_PATH` is also configured in `server.yaml` so it would be somewhat strange to have it configurable in two different ways (via `server.yaml` or `$XDG_CONFIG_HOME`)

Reply:
> I would prefer `~./mxcube/tmp` or `~./mxcube/` over the ie. `~./config/`.

It would be `~/.config/mxcube/mxcube-user.db` which is the modern convention. It can be overridden anyway, so you could set this to `~/.mxcube/mxcube-user-db` at ESRF via `USER_DB_PATH` in `server.yaml`.


> `USER_DB_PATH` is also configured in `server.yaml` so it would be somewhat strange to have it configurable in two different ways (via `server.yaml` or `$XDG_CONFIG_HOME`)

If I am not mistaken, `$XDG_CONFIG_HOME` gives you a way to override the default of `~/.config`. So if on your system you set `XDG_CONFIG_HOME=~/my-config` then `platformdirs.user_data_dir("mxcube")` would give you `~/my-config/mxcube` instead of `~/.config/mxcube`.

Reply:
Yes, I agree that it would be good in general. But I feel that it becomes a bit overkill to add an additional library to handle XDG directories.  What we want to do is to configure this path from `server.yaml`  and at the same time provide a safe default value. 

We can perhaps default the root path to `~/.config/mxcube/` if that supports the convention better

Reply:
> I feel that it becomes a bit overkill to add an additional library to handle XDG directories.

I agree. I guess we only support Linux, and the value can be overridden anyway, so no need for anything complex.

Reply:
Argh, my bad, I got my wires crossed, of course "user data" belongs in `~/.local/share/mxcube` not in `.config/mxcube`.

```suggestion
        str(Path.home() / ".local" / "share" / "mxcube" / "mxcube-user.db"), description=""
```

Do we need an empty description, though?

```suggestion
        str(Path.home() / ".local" / "share" / "mxcube" / "mxcube-user.db")
```

Reply:
Keeping description even if it's empty is better for consistency, not necessary but just follow good practice with Pydantic